src/server/system_services/system_server.js

```diff
@@ -985,30 +985,38 @@ async function update_endpoint_group(req) {
         .some(group => group.name === group_name);
 
     if (exists) {
+        const group = cluster.endpoint_groups.find(grp => grp.name === group_name);
         if (!_.isUndefined(is_remote)) {
-            const group = cluster.endpoint_groups.find(grp => grp.name === group_name);
             if (group.is_remote !== is_remote) {
                 // We do not throw in order to not fail the noobaa operator.
                 dbg.warn('update_endpoint_group: Conflicted is_remote value of ',
                     is_remote, ' for group: ', group, ' - aborting request');
                 return;
             }
+
         }
 
-        await system_store.make_changes({
-            update: {
-                clusters: [{
-                    $find: {
-                        _id: cluster._id,
-                        'endpoint_groups.name': group_name
-                    },
-                    $set: {
-                        'endpoint_groups.$.region': region,
-                        'endpoint_groups.$.endpoint_range': endpoint_range
-                    }
-                }]
-            }
-        });
+        // call make_changes only if there are actual changes to make.
+        // this check fixes a bug where make_changes sends a load_system_store notification 
+        // to the operator, which in its own reconcile sends back update_endpoint_group, and so forth
+        if (group.region !== region ||
+            group.endpoint_range.min !== endpoint_range.min ||
```

Comment:
I am not sure if `group.endpoint_range` is required or use defaults. please check that it is not null before accessing.

Also, I think using 
```
if (!_.deep_equal(group, { region, endpoint_range })) {
...
}
``` 
 is better solution in case we add more fields to the endpoint group

Reply:
I prefer to check explicitly. if we add more fields to the group, then the condition will always be false.